resolver/addressable_resolver.go

```diff
@@ -38,16 +38,21 @@ import (
 	"knative.dev/pkg/tracker"
 )
 
+type URIResolverFunc func(ctx context.Context, ref *corev1.ObjectReference) (*apis.URL, error)
```

Comment:
Please document `URIResolverFunc`

Reply:
I do wonder if we need a different return signature here, we can set the resolve up with more than one func, so the func needs to be able to say "I passed on this one, but it is not an error":

`(bool, *apis.URL, error)`

Then if the func resolves, it is `true, aURL, nil`
If the func passes it is `false, nil, nil`
If it hit a real error it is `true, nil, anError`

Reply:
sounds good.